client/db.go

```diff
@@ -314,6 +335,20 @@ type Tx struct {
 	txn *Txn
 }
 
+// SetSnapshotIsolation sets the transaction's isolation type to
+// snapshot. Transactions default to serializable isolation. The
+// isolation must be set before any operations are performed on the
```

Comment:
Doesn't matter when it's set as long as it's set before the EndTxn call goes through. Now that might not always be the case, but it's how the current implementation works.